src/Channel.ts

```diff
@@ -77,76 +77,79 @@ export class Channel extends EnhancedEventEmitter
 					this._recvBuffer.length + buffer.length);
 			}
 
-			if (this._recvBuffer!.length > NS_PAYLOAD_MAX_LEN)
+			if (this._recvBuffer.length > PAYLOAD_MAX_LEN)
 			{
-				logger.error('receiving buffer is full, discarding all data into it');
+				logger.error('receiving buffer is full, discarding all data in it');
 
 				// Reset the buffer and exit.
-				this._recvBuffer = undefined;
+				this._recvBuffer = Buffer.alloc(0);
 
 				return;
 			}
 
+			let msgStart = 0;
+
 			while (true) // eslint-disable-line no-constant-condition
 			{
-				let nsPayload;
+				const readLen = this._recvBuffer.length - msgStart;
 
-				try
+				if (readLen < 4)
 				{
-					nsPayload = netstring.nsPayload(this._recvBuffer);
+					// Incomplete data.
+					break;
 				}
-				catch (error)
-				{
-					logger.error(
-						'invalid netstring data received from the worker process: %s',
-						String(error));
 
-					// Reset the buffer and exit.
-					this._recvBuffer = undefined;
+				const dataView = new DataView(
+					this._recvBuffer.buffer,
+					this._recvBuffer.byteOffset + msgStart);
+				const msgLen = dataView.getUint32(0, littleEndian);
```

Comment:
To clarify: we always write the length of the message in the first 4 bytes of the data, right? In both Node/Rust and C++, am I right? And such a length does NOT include its own 4 bytes, right?